libs/fru.c

```diff
@@ -347,6 +348,27 @@ gettimeofday time
 #endif
 
 /**
+ * Calculate zero checksum for command header and FRU areas
+ */
+static
+uint8_t calc_checksum(void *blk, size_t blk_bytes)
+{
+    if (!blk || blk_bytes == 0) {
+        printf("Null pointer or zero buffer length\n");
```

Comment:
printf inside a library function is a really, really bad idea! add an extra parameter to the function to return the checksum and report status via the return value. Or better yet, do sanity checks on the caller's level. This function looks very much like a helper that could be (along with fru_area_checksum()) made static and unavailable to users.